pkg/v1/cli/command/core/config.go

```diff
@@ -190,12 +190,6 @@ func setEdition(cfg *configv1alpha1.ClientConfig, edition string) error {
 	switch editionOption {
 	case configv1alpha1.EditionCommunity, configv1alpha1.EditionStandard:
 		cfg.SetEditionSelector(editionOption)
-		// when community edition is set, configure the compatibility file to use
-		// community edition's.
-		err := cfg.SetCompatibilityFile(editionOption)
```

Comment:
This was removed from core-cli code and moved to individual plugin command invocations.

Reply:
Ok, now I'm completely out of my element, but please be patient with me.

I was curious about the change in logic about `SetCompatibilityFile`.  Looking at the change, I'm guessing that this compatibility file is only used when creating a cluster?  That is why you only need to trigger it for the `managed-cluster` and `cluster` plugins?

But somehow (probably me doing something wrong), the main configuration file does not get updated properly anymore.  This is a simplification of what I did:
```
# Check that the main config file is set to tkg (BOM and edition)
$ egrep 'bom|edition' ~/.config/tanzu/config.yaml
    bomRepo: projects-stg.registry.vmware.com/tkg
    edition: tkg

# Set the edition to TCE
$ tanzu config set cli.edition tce
# =====> Notice the BOM is no longer set properly
$ egrep 'bom|edition' ~/.config/tanzu/config.yaml
    bomRepo: projects-stg.registry.vmware.com/tkg
    edition: tce

# Remove the BOM line completely
$ grep -v bom ~/.config/tanzu/config.yaml > tt && \mv tt ~/.config/tanzu/config.yaml && grep bom ~/.config/tanzu/config.yaml

# Regenerate the config file by running any command
$ tanzu
Tanzu CLI
Usage:
  tanzu [command]
[...]
# =====> Notice the BOM is added but does not match the edition
$ egrep 'bom|edition' ~/.config/tanzu/config.yaml
    bomRepo: projects-stg.registry.vmware.com/tkg
    edition: tce
```
I don't know if this is actually a problem but it does not happen on the main branch.

Reply:
Thanks for catching that. I have fixed the problem and added unit test around `SetCompatibilityFileBasedOnEdition` function as well. 

Please note that `tanzu config set cli.edition tce` command will now just update the `edition` field in config and `BomRepo` and `Compatibility` information is updated when user runs `cluster` or `management-cluster` plugin command. Ideally, `BomRepo` and `Compatibility` information should not be part of `ClientConfig` at the first place and I have filled #3149 for that.